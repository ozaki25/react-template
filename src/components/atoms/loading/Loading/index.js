import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Loading = ({ show }) => <Container show={show} />;

const Container = styled.div`
  background-color: #000;
  border: none;
  cursor: wait;
  display: ${props => (props.show ? 'inherit' : 'none')};
  height: 100%;
  left: 0;
  margin: 0;
  opacity: 0.6;
  padding: 0;
  position: fixed;
  top: 0;
  width: 100%;
`;

Loading.propTypes = {
  show: PropTypes.bool,
};

Loading.defaultProps = {
  show: false,
};

export default Loading;
