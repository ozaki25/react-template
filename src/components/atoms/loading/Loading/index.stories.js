import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Loading from '.';

const stories = storiesOf('atoms/Loading', module);

stories.add(
  'show',
  withInfo('Lodging非表示')(() => (
    <div>
      <Loading show />
      <p>サンプル</p>
    </div>
  )),
);
stories.add(
  'hide',
  withInfo('Loading表示')(() => (
    <div>
      <Loading />
      <p>サンプル</p>
    </div>
  )),
);
