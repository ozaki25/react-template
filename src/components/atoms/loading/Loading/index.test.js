import React from 'react';
import { snapshot } from 'test/helpers';
import Loading from '.';

snapshot('Loading/show', <Loading show />);
snapshot('Loading/hide', <Loading />);
