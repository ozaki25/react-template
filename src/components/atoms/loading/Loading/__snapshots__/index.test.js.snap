// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Loading/hide 1`] = `
.c0 {
  background-color: #000;
  border: none;
  cursor: wait;
  display: none;
  height: 100%;
  left: 0;
  margin: 0;
  opacity: 0.6;
  padding: 0;
  position: fixed;
  top: 0;
  width: 100%;
}

<Loading
  show={false}
>
  <styled.div
    show={false}
  >
    <div
      className="c0"
    />
  </styled.div>
</Loading>
`;

exports[`should match snapshot: Loading/show 1`] = `
.c0 {
  background-color: #000;
  border: none;
  cursor: wait;
  display: inherit;
  height: 100%;
  left: 0;
  margin: 0;
  opacity: 0.6;
  padding: 0;
  position: fixed;
  top: 0;
  width: 100%;
}

<Loading
  show={true}
>
  <styled.div
    show={true}
  >
    <div
      className="c0"
    />
  </styled.div>
</Loading>
`;
